src/lex/cursor.rs

```diff
@@ -68,10 +68,10 @@ impl<'a> Cursor<'a> {
 
     pub(crate) fn next_line_remaining(&mut self) -> &'a str {
         let spaces = self.source[self.index..]
-            .find(|c: char| !c.is_whitespace())
+            .find(|c: char| c != ' ')
             .unwrap_or(self.source.len());
 
-        self.col += spaces;
+        self.col += 1;
```

Comment:
That's really a problem, many characters are defined as whitespace, so I'm looking for a solution to make this check as useful as possible. is_whitespace() method seems will drop some special contents.

Reply:
I found that the extraction process is too redundant, and it can be written as:
1. Get a slice of string from now to the next new line character.
2. Trim it whose start and end.
3. Seek to next line.

Reply:
Yes, I found remove the '!' of `!c.is_whitespace()` while leave the whitespace in final string, so just trim it and return it, that could be the better way and less redundant code.